src/include/optimizer/optimizer.h

```diff
@@ -52,11 +53,14 @@ class Optimizer : public AbstractOptimizer {
 
   Optimizer();
 
+
   std::shared_ptr<planner::AbstractPlan> BuildPelotonPlanTree(
       const std::unique_ptr<parser::SQLStatementList> &parse_tree) override;
 
   void Reset() override;
 
+  concurrency::Transaction *txn = nullptr;
```

Comment:
I can modify txn to argument in optimizer. And please give me a reason why do you hate member variable so much?